src/views/CTLViewsHoC.jsx

```diff
@@ -70,9 +77,11 @@ const CTLViewsHoC = (WrappedView) => {
 				const { prettyUrlName } = getListingInfo.payload;
 
 				setStateListingInfo(getListingInfo.payload);
-
 				// Redirect to pretty url if one exists for listing info
-				if (prettyUrlName) {
+				if (idType()) {
```

Comment:
A redirect shouldn't be happening in this case. Redirect only happens if `c-code` is used and `prettyUrlName` exists. You just have to add a negation on the "idType" `!idType()` to line 84 and remove this one like so...
```js
if (prettyUrlName && !idType()) {
  setHasBeenRedirected(true);
	const queryString = appendOrUpdateToQueryString(
	  search,
		'redirect',
		'true'
  );
	navigate(
	  `${CodeOrPurlPath({ codeOrPurl: prettyUrlName })}${queryString}`
	);
}
```

Reply:
Now even wondering if the function you have above is even necessary. Why not just do
`prettyUrlName && paramType !== 'code'` and do without the function `idType`